### PR TITLE
Fix no-dso (shlibloadtest)

### PR DESCRIPTION
**Pull request by levitte**
##### Checklist

- [x] CLA is signed

### CODE REVIEW

File: test/shlibloadtest.c [hunk 1]
```diff
@@ -12,6 +12,8 @@
 #include <stdlib.h>
 #include <openssl/opensslv.h>
 
+#ifdef DSO_DLFCN
```
Comment (mattcaswell) [inline test/shlibloadtest.c]: I don't think this fix is quite right. The test also handles DSO_WIN32

Comment (levitte) [inline test/shlibloadtest.c]: Argh.  'k hold on

**Comment by levitte** [comment]
New commit with a hopefully better way.  Makes more sense?

**Review by mattcaswell — APPROVED** [review]
I will merge.

**Comment by mattcaswell** [comment]
Author is set as richard@levitte.org, not levitte@openssl.org....is that correct???

**Comment by levitte** [comment]
I'll fix.  Forgot to change my user name in that checkout...

**Comment by levitte** [comment]
Fixed

**Comment by mattcaswell** [comment]
Merged.

**Pull request closed without merging**